java/system-test/system-test-suite/src/test/java/sleeper/systemtest/suite/CompactionIT.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.systemtest.suite;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import sleeper.compaction.strategy.impl.BasicCompactionStrategy;
+import sleeper.core.partition.PartitionTree;
+import sleeper.core.statestore.FileReference;
+import sleeper.core.statestore.FileReferenceFactory;
+import sleeper.systemtest.datageneration.RecordNumbers;
+import sleeper.systemtest.suite.dsl.SleeperSystemTest;
+import sleeper.systemtest.suite.dsl.reports.SystemTestReports;
+import sleeper.systemtest.suite.testutil.AfterTestPurgeQueues;
+import sleeper.systemtest.suite.testutil.AfterTestReports;
+import sleeper.systemtest.suite.testutil.SystemTest;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.LongStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.COMPACTION_JOB_QUEUE_URL;
+import static sleeper.configuration.properties.table.TableProperty.COMPACTION_FILES_BATCH_SIZE;
+import static sleeper.configuration.properties.table.TableProperty.COMPACTION_STRATEGY_CLASS;
+import static sleeper.core.testutils.printers.FileReferencePrinter.printFiles;
+import static sleeper.systemtest.suite.fixtures.SystemTestInstance.MAIN;
+import static sleeper.systemtest.suite.testutil.PartitionsTestHelper.partitionsBuilder;
+
+@SystemTest
+public class CompactionIT {
+    @TempDir
+    private Path tempDir;
+    private FileReferenceFactory factory;
+    private PartitionTree initialPartitions;
+
+    @BeforeEach
+    void setUp(SleeperSystemTest sleeper, AfterTestReports reporting, AfterTestPurgeQueues purgeQueues) {
+        sleeper.connectToInstance(MAIN);
+        reporting.reportIfTestFailed(SystemTestReports.SystemTestBuilder::compactionTasksAndJobs);
+        purgeQueues.purgeIfTestFailed(COMPACTION_JOB_QUEUE_URL);
+        initialPartitions = partitionsBuilder(sleeper).singlePartition("root").buildTree();
+        factory = FileReferenceFactory.from(initialPartitions);
+    }
```

Comment:
Maybe the main thing we want to test here is that the normal compaction strategy can handle a normal situation where know we'd want it to do a compaction. I think the distinction between the different test cases here is covered by the normal unit and integration tests.

Can we write a passing scenario without setting it to use the basic compaction strategy? This can be similar to CompactionPerformanceIT but with a lot fewer records and files.